### ISSUE TITLE
Difference between concat and plain pipeing

### ISSUE THREAD

**Issue by LostKobrakai** [plataformatec/nimble_parsec#7]
I've not much experience with parsers, but given a super simple markdown headline like `# Heading` I'm wondering why the first following example doesn't work, while the latter one does:

```elixir
  heading_1 =
    string("#")
    |> ignore()
    |> utf8_string([], min: 1)

  defparsec(:markdown, heading_1)
```

```elixir
  heading_1 =
    string("#")
    |> ignore()
    |> concat(utf8_string([], min: 1))

  defparsec(:markdown, heading_1)
```

**Comment by josevalim** [comment]
What do you mean by it does not work? Isto a compilation error? Are you
passing some input and it does not work?

Please provide as much information as you can. Say what happened and what
you expected.
-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

**Comment by LostKobrakai** [comment]
Code example 1 does error with the following info for the input `# Heading`:
`{:error, "expected string \"#\", followed by utf8 codepoint", " Heading", %{}, {1, 0}, 1}``
while code example two does not error.

I had expected the first code example to work and I'm not really sure why it would remove the initial `#` and then try again to match for another `#`.

**Comment by josevalim** [comment]
@LostKobrakai seems like a bug. Investigating.

**Comment by josevalim** [comment]
Fixed and release v0.2.1. Thank you for the report.
